### PR TITLE
refactor(backend): use LogError on contributionResolver

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

fix logger on contributionResolver

### Issues

- contributes to #2487 
- contributes to #1870 
- based on #2663

### Todo

- [X] None

### CODE REVIEW

File: backend/src/graphql/resolver/ContributionLinkResolver.test.ts [hunk 1]
```diff
@@ -246,6 +246,7 @@ describe('Contribution Links', () => {
         })
 
         it('returns an error if missing startDate', async () => {
+          jest.clearAllMocks()
```
Comment (Elweyn) [inline backend/src/graphql/resolver/ContributionLinkResolver.test.ts:249]: I'm not sure why you add this? 
If it is to be sure that only the right error is thrown and logged, why don't you add it on every tests that checks on Throws and Logs? For example line 504 and 584 would then also need it.

File: backend/src/graphql/resolver/ContributionLinkResolver.ts [hunk 2]
```diff
@@ -39,35 +40,30 @@ export class ContributionLinkResolver {
     }: ContributionLinkArgs,
   ): Promise<ContributionLink> {
     isStartEndDateValid(validFrom, validTo)
+    // TODO: this should be enforced by the schema.
```
Comment (Elweyn) [inline backend/src/graphql/resolver/ContributionLinkResolver.ts:43]: Ticket?

File: backend/src/graphql/resolver/ContributionLinkResolver.ts [hunk 3]
```diff
@@ -39,35 +40,30 @@ export class ContributionLinkResolver {
...
-      const msg = `The value of 'name' with a length of ${name.length} did not fulfill the requested bounderies min=${CONTRIBUTIONLINK_NAME_MIN_CHARS} and max=${CONTRIBUTIONLINK_NAME_MAX_CHARS}`
-      logger.error(`${msg}`)
-      throw new Error(`${msg}`)
+    if (name.length < CONTRIBUTIONLINK_NAME_MIN_CHARS) {
+      throw new LogError('The value of name is too short', name.length)
     }
+    if (name.length > CONTRIBUTIONLINK_NAME_MAX_CHARS) {
+      throw new LogError('The value of name is too long', name.length)
+    }
+    // TODO: this should be enforced by the schema.
```
Comment (Elweyn) [inline backend/src/graphql/resolver/ContributionLinkResolver.ts:53]: Ticket?

File: backend/src/graphql/resolver/ContributionResolver.test.ts [hunk 4]
```diff
@@ -422,32 +423,33 @@ describe('ContributionResolver', () => {
         resetToken()
       })
 
-      describe('wrong contribution id', () => {
```
Comment (Elweyn) [inline backend/src/graphql/resolver/ContributionResolver.test.ts:425]: Don't understand why you need to change the order makes it harder to read the changes.
I first thought you removed it!

File: backend/src/graphql/resolver/ContributionResolver.ts [hunk 5]
```diff
@@ -403,18 +389,11 @@ export class ContributionResolver {
       withDeleted: true,
       relations: ['user'],
     })
-    if (!emailContact) {
-      logger.error(`Could not find UserContact with email: ${email}`)
-      throw new Error(`Could not find UserContact with email: ${email}`)
+    if (!emailContact || !emailContact.user) {
+      throw new LogError('Could not find User', email)
     }
-    const user = emailContact.user
```
Comment (Elweyn) [inline backend/src/graphql/resolver/ContributionResolver.ts:410]: Why remove this after check if emailContact.deletedAt we never need emailContact but the user in it!

**Review by Elweyn — CHANGES_REQUESTED** [review]

File: backend/src/graphql/resolver/ContributionResolver.test.ts (hunk 4 above)
Comment (ulfgebhardt) [reply backend/src/graphql/resolver/ContributionResolver.test.ts:425]: Because thats how the code is
image

Comment (Elweyn) [reply backend/src/graphql/resolver/ContributionResolver.test.ts:425]: OK!

File: backend/src/graphql/resolver/ContributionLinkResolver.ts (hunk 3 above)
Comment (Mogge) [inline backend/src/graphql/resolver/ContributionLinkResolver.ts:53]: Same. Test not needed

File: backend/src/graphql/resolver/ContributionMessageResolver.ts [hunk 6]
```diff
@@ -54,8 +54,7 @@ export class ContributionMessageResolver {
       await queryRunner.commitTransaction()
     } catch (e) {
       await queryRunner.rollbackTransaction()
-      logger.error(`ContributionMessage was not successful: ${e}`)
-      throw new Error(`ContributionMessage was not successful: ${e}`)
+      throw new LogError('ContributionMessage was not sent successfully', e)
```
Comment (Mogge) [inline backend/src/graphql/resolver/ContributionMessageResolver.ts:57]: Do we want to give the reason to the frontend as well? See contribution message resolver

File: backend/src/graphql/resolver/ContributionResolver.test.ts [hunk 7]
```diff
@@ -521,22 +522,21 @@ describe('ContributionResolver', () => {
...
           expect(logger.error).toBeCalledWith(
-            'user of the pending contribution and send user does not correspond',
+            'Can not update contribution of another user',
+            expect.any(Object),
+            expect.any(Number),
           )
         })
       })
 
+      // TODO: why is this here - this is a different call `adminUpdateContribution` not `updateContribution`
```
Comment (Mogge) [inline backend/src/graphql/resolver/ContributionResolver.test.ts:539]: Where do you want it? May be move it there and get rid of to do?

File: backend/src/seeds/users/stephen-hawking.ts [hunk 8]
```diff
@@ -1,5 +1,6 @@
 import { UserInterface } from './UserInterface'
 
+// TODO: the generated email_contact is not deleted
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/seeds/users/stephen-hawking.ts:3]: I agree. If not this comment will stay here for ever and nothing will happen

**Review by Mogge — CHANGES_REQUESTED** [review]
In general: as far as I know yo write `cannot` in English and not `can not`

File: backend/src/graphql/resolver/ContributionResolver.ts (hunk 5 above)
Comment (ulfgebhardt) [reply backend/src/graphql/resolver/ContributionResolver.ts:410]: this way is more memory efficient. else we have two copies of the user object in memory.

File: backend/src/seeds/users/stephen-hawking.ts (hunk 8 above)
Comment (ulfgebhardt) [inline backend/src/seeds/users/stephen-hawking.ts:3]:
```suggestion
```

[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply backend/src/seeds/users/stephen-hawking.ts]: https://github.com/gradido/gradido/issues/2687

File: backend/src/graphql/resolver/ContributionResolver.test.ts (hunk 7 above)
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/ContributionResolver.test.ts:539]:
```suggestion
```

**Review by Elweyn — APPROVED** [review]

**Pull request merged**